### PR TITLE
[exporter/sumologicexporter]: adds support for tracing

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

adds support for tracing

It should be the last PR which moves code from Sumo Logic repository. I will compare the codebase after merging it

**Link to tracking Issue:**

#32315

**Testing:**

* unit tests

**Documentation:**

* Readme

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I hope this is the last one. I am going to be very happy about closing this saga 🤞

File: exporter/sumologicexporter/config.go
Comment (swiatekm-sumo) [inline exporter/sumologicexporter/config.go:40]: This isn't correct. Rather, the way the logs are presented in Sumo will depend on the record attributes and the data type of Body. 
* If record attributes are empty, and the body is a string, we show plain text. 
* If record attributes are empty, and the body is a structured data type, we show the JSON equivalent of body.
* If they're not, we show a JSON object with the log body in the "body" key, and the rest of record attributes as the rest of the keys.

To be honest, this is complicated enough that it should really be a README section, or we can link to the Sumo documentation.

Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go]: I linked doucumentation in `README.md` and `config.go`

**Review by swiatekm — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
cc @songy23 @dmitryax @andrzej-stencel

File: .chloggen/drosiek-tracing.yaml
Comment (songy23) [inline .chloggen/drosiek-tracing.yaml:4]: Why is it a breaking change?

File: exporter/sumologicexporter/exporter.go
Comment (songy23) [inline exporter/sumologicexporter/exporter.go:402]: I'm not familiar with Sumo but why does it need to create a new sender for each trace batch, why not create a sender once and reuse it?

File: .chloggen/drosiek-tracing.yaml
Comment (sumo-drosiek) [reply .chloggen/drosiek-tracing.yaml:4]:
```suggestion
change_type: enhancement
```

File: exporter/sumologicexporter/exporter.go
Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go:402]: We used to have implementation which would required synchronisation before, but now I think we could reuse it 🤔

I created issue for it: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33105

**Review by songy23 — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
cc: @collector-contrib-approvers

File: exporter/sumologicexporter/config.go
Comment (crobert-1) [inline exporter/sumologicexporter/config.go:56]: Do you want to add a README entry for this option? 

Side note, if there's only one valid, is it worth having a config option? (I guess this depends on whether there are plans to support more formats, I'll defer to you and other code owners here as to usefulness.)

File: exporter/sumologicexporter/config.go
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:173]: Updated

File: exporter/sumologicexporter/config.go
Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go]: I just removed it. It seems we do not use it anywhere, so should be safe to remove. We can add it in the future if we need to

File: exporter/sumologicexporter/README.md
Comment (crobert-1) [inline exporter/sumologicexporter/README.md:104]: Is this maximum connection timeout ever enforced or tested? It's unrelated to this PR, so no need to do anything now, just something to consider if it's not currently enforced.

**Review by crobert-1 — APPROVED** [review]

File: exporter/sumologicexporter/README.md
Comment (sumo-drosiek) [reply exporter/sumologicexporter/README.md:104]: I don't think we test it
Created issue so we can keep track of it: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33151

**Pull request merged**
